File: src/test/generate_test_data/generateRunSaveTest_EMGrank.m
function[] = generateRunSaveTest_EMGrank(n, p, m, k, mini, maxi, gamma, rank, varargin)

	%set defaults for optional inputs
	optargs = {200 15 10 3 5 10 1.0 1:3};
	%replace defaults by user parameters
	optargs(1:length(varargin)) = varargin;
	[n, p, m, k, mini, maxi, gamma, rank] = optargs{:};
	mini = int64(mini);
	maxi = int64(maxi);
	rank = int64(rank);
	tau = 1e-6;

	Pi = (1.0/k)*ones(1,k);
	Rho = zeros(m,m,k);
	for r=1:k
		Rho(:,:,r) = eye(m);
	end

	%Generate X and Y
	[X, Y, ~] = generateIOdefault(n, p, m, k);

	%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

	testFolder = 'data/';
	mkdir(testFolder);
	delimiter = ' ';

	%save inputs
	dlmwrite(strcat(testFolder,'Rho'), reshape(Rho,1,[]), delimiter);
	dlmwrite(strcat(testFolder,'Pi'), Pi, delimiter);
	dlmwrite(strcat(testFolder,'mini'), mini, delimiter);
	dlmwrite(strcat(testFolder,'maxi'), maxi, delimiter);
	dlmwrite(strcat(testFolder,'X'), reshape(X,1,[]), delimiter);
	dlmwrite(strcat(testFolder,'Y'), reshape(Y,1,[]), delimiter);
	dlmwrite(strcat(testFolder,'tau'), tau, delimiter);
	dlmwrite(strcat(testFolder,'rank'), rank, delimiter);
	dlmwrite(strcat(testFolder,'dimensions'), [n,p,m,k], delimiter);;

	[phi,LLF] = EMGrank(Pi,Rho,mini,maxi,X,Y,tau,rank);

	%save output
	dlmwrite(strcat(testFolder,'phi'), reshape(phi,1,[]), delimiter);
	dlmwrite(strcat(testFolder,'LLF'), LLF, delimiter);

end

File: src/test/generate_test_data/generateRunSaveTest_EMGrank.R
generateRunSaveTest_EMGrank = function(n=200, p=15, m=10, k=3, mini=5, maxi=10, gamma=1.0, rank = c(1,2,4)){
  testFolder = "data/"
  dir.create(testFolder, showWarnings=FALSE, mode="0755")
  delimiter = " "
  
  tau = 1e-6
  
  pi = rep(1.0/k, k)
  rho = array(0, dim=c(m,m,k))
  
  for(i in 1:k){
    rho[,,i] = diag(1,m)
  }

  #Generate X and Y
	require(valse)
  generateIOdef = valse:::generateIOdefault(n, p, m, k)
  
  #save inputs
  write.table(paste(testFolder,"rho",sep=""), rho, sep=delimiter)
  write.table(paste(testFolder,"pi",sep=""), pi, sep=delimiter)
  write.table(paste(testFolder,"mini",sep=""), mini, sep=delimiter)
  write.table(paste(testFolder,"maxi",sep=""), maxi, sep=delimiter)
  write.table(paste(testFolder,"X",sep=""), generateIOdef$X sep=delimiter)
  write.table(paste(testFolder,"Y",sep=""), generateIOdef$Y, sep=delimiter)
  write.table(paste(testFolder,"tau",sep=""), tau, sep=delimiter)
  write.table(paste(testFolder,"rank",sep=""), rank, sep=delimiter)
  write.table(paste(testFolder,"dimensions",sep=""), c(n,p,m,k), sep=delimiter)
  
  EMG_rank = EMG(pi,rho,mini,maxi,X,Y,tau,rank)
  
  #save output
  write.table(paste(testFolder,"phi",sep=""), EMG_rank$phi, sep=delimiter)
  write.table(paste(testFolder,"LLF",sep=""), EMG_rank$LLF, sep=delimiter)
}
